File: 3. Arrays and String/28. Find the Index of the First Occurence of a String/first_occurence.cpp
class Solution
{
public:
    int strStr(string haystack, string needle)
    {
        // if length if haystack < length of needle return -1
        if (haystack.length() < needle.length())
        {
            return -1;
        }

        // for each of the index in haystack
        // find needle
        for (int i = 0; i <= haystack.length() - needle.length(); ++i)
        {
            // if needle found return index
            if (haystack.substr(i, needle.length()) == needle)
            {
                return i;
            }
        }
        // if not found return -1
        return -1;
    }
};

File: 3. Arrays and String/28. Find the Index of the First Occurence of a String/first_occurence.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        hay_len = len(haystack)
        ned_len = len(needle)

        # if lenth of haystack id less than needle return -1
        if hay_len < ned_len:
            return -1

        # find needle in haystack iteratively
        for i in range(hay_len - ned_len + 1):
            # if neele found return index
            if haystack[i : i + ned_len] == needle:
                return i

        # if not found return -1
        return -1
